TELCODOCS-1109 (#67822)
openshift/openshift-docs: modules/kmm-hub-using-the-managedclustermodule.adoc

// Module included in the following assemblies:
//
// * hardware_enablement/kmm-kernel-module-management.adoc

:_content-type: PROCEDURE
[id="kmm-hub-using-the-managedclustermodule_{context}"]
= Using the `ManagedClusterModule` CRD

Use the `ManagedClusterModule` Custom Resource Definition (CRD) to configure the deployment of kernel modules on spoke clusters.

> This CRD is cluster-scoped, wraps a ``Module`` spec and adds the following additional fields:

Author: Changed.

[source,yaml]
----
apiVersion: hub.kmm.sigs.x-k8s.io/v1beta1
kind: ManagedClusterModule
metadata:
  name: <my-mcm>
  # No namespace, because this resource is cluster-scoped.
spec:
  moduleSpec: <1>
    selector: <2>
      node-wants-my-mcm: 'true'

  spokeNamespace: <some-namespace> <3>

  selector:

>     wants-my-mcm: 'true'  # Selects ManagedCluster objects

Author: Changed to a call-out.

----

> <1> `moduleSpec`: Contains `moduleLoader` and `devicePlugin` sections, just like in a `Module` resource.

Reviewer: ```suggestion
<1> Contains the `moduleLoader` and `devicePlugin` sections, similar to a `Module` resource.
```
Do we want to show the `devicePlugin`? 
What about the spaces (Line 23 and 25? Should they be removed?

Author: Changed "just like in...". No other changes.

> <2> Selects nodes within the `ManagedCluster`.

Reviewer: ```suggestion
<2> Selects nodes within the managed cluster.
```

Author: No change.
